aws/resource_aws_elasticsearch_domain_policy.go

```diff
@@ -40,7 +40,7 @@ func resourceAwsElasticSearchDomainPolicyRead(d *schema.ResourceData, meta inter
 		DomainName: aws.String(name),
 	})
 	if err != nil {
-		if awsErr, ok := err.(awserr.Error); ok && awsErr.Code() == "ResourceNotFound" {
+		if awsErr, ok := err.(awserr.Error); ok && awsErr.Code() == "ResourceNotFoundException" {
```

Comment:
This was already addressed in https://github.com/terraform-providers/terraform-provider-aws/pull/2249